### ISSUE TITLE
Feature Request: Customizable date/timestamps

### ISSUE THREAD

**Issue by KatieTheDev** [panr/hugo-theme-terminal#493]
Hi all,

While trying to customize my page to my liking, I discovered that Hugo has built-in support for different date and time formats. I would like to suggest a new feature where we can set if a timestamp should be shown or only date. I have a proof of concept available in my [website's repository](https://github.com/KatieTheDev/katiethe.dev/blob/main/layouts/_default/single.html). I believe that using some if-statements and adding some options to the main config file, that we should be able to add this as a feature.

If the community finds interest in this, I would be more than happy to start work on a PR.

**Comment by KatieTheDev** [comment]
![image](https://github.com/user-attachments/assets/6088f280-7395-4b1b-8ee9-8a9dfaaa518f)
This is an example of it working in action with how I currently have it set up.

**Comment by KatieTheDev** [comment]
In issue #477 another user has suggested replacing the `{{- .Date.Format "2006-01-02" -}}` string with a time.Format :date_medium string. I would be interested in incorporating this into a potential PR, though I am unsure if it would be better with it separate.

**Comment by KatieTheDev** [comment]
I should add that under my implementation, extra CSS is required to make it look how I did. In order to make the AM/PM designation uppercase, I created the following CSS rule:
```css
.post-time {
  text-transform: uppercase;
  padding: 0px;
  margin: 0px;
  border: 0px;
}
```
In order to fix the issue of the extra span created for .post-time, I added the following css as well:
```css
.post-time {
  text-transform: uppercase;
  padding: 0px;
  margin: 0px;
  border: 0px;
}
```
All of this can be found in my /assets/css/style.css file in my website's repository.

**Comment by KatieTheDev** [comment]
As such, a PR should add these settings to one of the main CSS files. I think terminal.css would likely work well, unless any maintainers have suggestions?

Also, my apologies for the many comments. This is how I think and it is immensely helpful for me to be able to break up my thoughts this way.

**Comment by panr** [comment]
> If the community finds interest in this, I would be more than happy to start work on a PR.

Hi @KatieTheDev, you are more than welcome to start the PR ;-) This seems to be a very interesting feature as long as we keep it true to the official Hugo docs (we don't want to add unnecessary abstractions).

The layout string probably should be configurable in `config.toml`, but I suggest to keep the current date format as default (as fallback when the new option is empty/null etc.) to not introducing breaking changes to anyone (at least for now).

Is it possible to mix (concatenate) two layouts like this: `{{ .Date | .time.Format ":date_medium :time_short" }}`? I haven't played with it yet?

**Comment by KatieTheDev** [comment]
> > If the community finds interest in this, I would be more than happy to start work on a PR.
> 
> Hi @KatieTheDev, you are more than welcome to start the PR ;-) This seems to be a very interesting feature as long as we keep it true to the official Hugo docs (we don't want to add unnecessary abstractions).
> 
> The layout string probably should be configurable in `config.toml`, but I suggest to keep the current date format as default (as fallback when the new option is empty/null etc.) to not introducing breaking changes to anyone (at least for now).
> 
> Is it possible to mix (concatenate) two layouts like this: `{{ .Date | .time.Format ":date_medium :time_short" }}`? I haven't played with it yet?

I am not sure if that is possible, so possibly my PR could be refactored somewhat if it is. I will test this and find out. That said, I have it setup so that if my new configuration options are not set to true, it will default back to the existing datestamp. Please check out PR #494 as it has my work so far.

**Comment by KatieTheDev** [comment]
Actually with how I have it set up to where you can enable/disable seconds, it may be best to keep them as separate lines so that we aren't duplicating code?

**Comment by KatieTheDev** [comment]
Closed as discussion fits better in PR #494.
